### PR TITLE
Bundle all activityPub in middleware

### PR DESCRIPTION
**Pull request by roschaefer**
This way we can disable it for the seeder server which does not need to
create RSA keys for seeded users.

### CODE REVIEW

**Comment by roschaefer** [comment]
@Mastercuber can you give me a review on this? Ignore the failing FOSSA test.

**Comment by roschaefer** [comment]
@appinteractive Can you tell me why FOSSA fails? I did not add any dependency, so why is it failing?

I cannot see why it is failing because FOSSA does not recognize my maintainer role *even when I am logged in*:
Screenshot (41)

**Comment by mattwr18** [comment]
superficiously it looks good, but and it's passing the build, so I'm tempted to approve, but it's probably better to wait for @Mastercuber to review it.

Also, is this still `[WIP]`, or have you maybe forgotten to delete that?

**Review by Mastercuber — COMMENTED** [review]
One literal  to change and it's done

File: src/activitypub/security/httpSignature.spec.js [hunk 1]
```diff
@@ -1,69 +1,84 @@
...
-      const signer = crypto.createSign('rsa-sha256')
-      signer.update('(request-target): post /activitypub/users/max/inbox\ndate: 2019-03-08T14:35:45.759Z\nhost: democracy-app.de\ncontent-type: application/json')
-      signatureB64 = signer.sign({ key: user.privateKey, passphrase: 'a7dsf78sadg87ad87sfagsadg78' }, 'base64')
-    })
-    it('creates a Signature with given privateKey, keyId, url and headers (default algorithm: "rsa-sha256")', () => {
-      const httpSignature = createSignature(user.privateKey, 'https://human-connection.org/activitypub/users/lea#main-key', 'https://democracy-app.de/activitypub/users/max/inbox', headers)
+      beforeEach(() => {
+        const signer = crypto.createSign('rsa-sha256')
+        signer.update('(request-target): post /activitypub/users/max/inbox\ndate: 2019-03-08T14:35:45.759Z\nhost: democracy-app.de\ncontent-type: application/json')
+        signatureB64 = signer.sign({ key: privateKey, passphrase: 'a7dsf78sadg87ad87sfagsadg78' }, 'base64')
```
[the remark this answers is not in the document]
Comment (roschaefer) [reply src/activitypub/security/httpSignature.spec.js]: done

**Pull request merged**